packages/cli/src/commands/proxy/utils.ts

```diff
@@ -0,0 +1,147 @@
+import { existsSync } from 'node:fs';
+import fs from 'node:fs/promises';
+import type http from 'node:http';
+import path from 'node:path';
+
+import { isValidUrl } from '@logto/core-kit';
+import { conditional, trySafe } from '@silverhand/essentials';
+import chalk from 'chalk';
+import { createProxyMiddleware, responseInterceptor } from 'http-proxy-middleware';
+import { type OnProxyEvent } from 'http-proxy-middleware/dist/types.js';
+import mime from 'mime';
+
+import { consoleLog } from '../../utils.js';
+
+import { type ProxyResponseHandler } from './types.js';
+
+export const createProxy = (targetUrl: string, onProxyResponse?: OnProxyEvent['proxyRes']) => {
+  const hasResponseHandler = Boolean(onProxyResponse);
+  return createProxyMiddleware({
+    target: targetUrl,
+    changeOrigin: true,
+    selfHandleResponse: hasResponseHandler,
+    ...conditional(
+      hasResponseHandler && {
+        on: {
+          proxyRes: onProxyResponse,
+        },
+      }
+    ),
+  });
+};
+
+export const createStaticFileProxy = (staticPath: string) => {
+  const index = 'index.html';
+  const indexContentType = 'text/html; charset=utf-8';
+  const noCache = 'no-cache, no-store, must-revalidate';
+  const maxAgeSevenDays = 'max-age=604_800_000';
+
+  return async (request: http.IncomingMessage, response: http.ServerResponse) => {
+    if (!request.url) {
+      response.writeHead(400).end();
+      return;
+    }
+
+    if (request.method === 'HEAD' || request.method === 'GET') {
+      const loadIndex = !isFileAssetPath(request.url);
```

Comment:
nit: includes `index.html`

Reply:
Yeah, I'm aware of this, but it doesn't hurt.

Reply:
then it should be `fallBackToIndex`?

Reply:
Updated